From 607b1db4c388e25e7cf340d8928c12a3f9f36235 Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Thu, 19 Jul 2018 15:03:29 +0800
Subject: [PATCH] Check whether guest user is already created or not

CarUserManagerHelper part:
Car settings is using CarUserManagerHelper, implement
findCurrentGuestUser() api in CarUserManagerHelper.

Change-Id: Ib96aec70710a4d083c018d77139a04cb06d30a6d
Tracked-On: OAM-72293
Signed-off-by: robinz1x <robinx.zhang@intel.com>
Signed-off-by: Yan, WalterX <walterx.yan@intel.com>
---
 car-lib/src/android/car/user/CarUserManagerHelper.java | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/car-lib/src/android/car/user/CarUserManagerHelper.java b/car-lib/src/android/car/user/CarUserManagerHelper.java
index 4a69f0b..6119a39 100644
--- a/car-lib/src/android/car/user/CarUserManagerHelper.java
+++ b/car-lib/src/android/car/user/CarUserManagerHelper.java
@@ -481,6 +481,21 @@ public class CarUserManagerHelper {
     }
 
     /**
+     * Find the current guest user.
+     */
+    public UserInfo findCurrentGuestUser() {
+        final List<UserInfo> users = getAllUsers();
+        final int size = users.size();
+        for (int i = 0; i < size; i++) {
+            final UserInfo user = users.get(i);
+            if (user.isGuest() && !user.guestToRemove) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    /**
      * Gets a bitmap representing the user's default avatar.
      *
      * @param userInfo User whose avatar should be returned.
-- 
1.9.1

